UD-Viz-Core/src/Modules/CityObjects/ViewModel/CityObjectProvider.js

```diff
@@ -191,21 +191,22 @@ export class CityObjectProvider extends EventSender {
    * @private
    */
   _updateTilesManager() {
-    this.tilesManager.update();
-    this.tilesManager.removeAllStyles();
-
-    if (this.layer === undefined) {
-      this.layerCityObjectIds = [];
-    } else {
-      this.layerCityObjectIds = this.tilesManager
-        .findAllCityObjects(this.layer.filter.accepts)
-        .map((co) => co.cityObjectId);
-      
-      this.tilesManager.setStyle(this.layerCityObjectIds, this.layer.style);
-    }
-
-    if (!!this.selectedCityObjectId) {
-      this.tilesManager.setStyle(this.selectedCityObjectId, this.defaultSelectionStyle);
+    this.layerManager.update3DTiles();
+    this.layerManager.removeAll3DTilesStyles();
+    if (!!this.selectedCityObject) {
+      let tileManager = this.layerManager.getTilesManagerByLayerID(this.selectedCityObject.tile.layer.id);
+
+      if (this.layer === undefined) {
```

Comment:
`this.layer` is the layer corresponding to the current selected city object. Since we can select an object that comes from another layer, we need to change the selected object but also the corresponding layer